engine/access/rpc/backend/script_executor_test.go

```diff
@@ -0,0 +1,287 @@
+	db := pebbleStorage.NewBootstrappedRegistersWithPathForTest(s.T(), s.dbDir, s.height, s.height)
+	pebbleRegisters, err := pebbleStorage.NewRegisters(db)
+	s.Require().NoError(err)
+	s.registerIndex = pebbleRegisters
+
+	derivedChainData, err := derived.NewDerivedChainData(derived.DefaultDerivedDataCacheSize)
+	s.Require().NoError(err)
+
+	indexerCore, err := indexer.New(
+		s.log,
+		metrics.NewNoopCollector(),
+		nil,
+		s.registerIndex,
+		s.headers,
+		nil,
+		nil,
+		nil,
+		nil,
+		s.chain,
+		derivedChainData,
+		nil,
+	)
+	s.Require().NoError(err)
+
+	s.scripts = execution.NewScripts(
+		s.log,
+		metrics.NewNoopCollector(),
+		s.chain.ChainID(),
+		entropyBlock,
+		s.headers,
+		indexerCore.RegisterValue,
+		query.NewDefaultConfig(),
+		derivedChainData,
+		true,
+	)
+	s.bootstrap()
+}
+
+func (s *ScriptExecutorSuite) TestExecuteAtBlockHeight() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	script := []byte("access(all) fun main() { }")
+	var scriptArgs [][]byte
+	var expectedResult = []byte("{\"type\":\"Void\"}\n")
+
+	s.Run("test scrypt execution without version control", func() {
+		scriptExec := NewScriptExecutor(s.log, nil, uint64(0), math.MaxUint64)
+		s.reporter.On("HighestIndexedHeight").Return(s.height+1, nil)
+
+		err := scriptExec.Initialize(s.indexReporter, s.scripts)
+		s.Require().NoError(err)
+
+		res, err := scriptExec.ExecuteAtBlockHeight(ctx, script, scriptArgs, s.height)
+		s.Assert().NoError(err)
+		s.Assert().NotNil(res)
+		s.Assert().Equal(expectedResult, res)
+	})
+
+	s.Run("test scrypt execution with version control with compatible version", func() {
+		// Set up a mock version beacons storage.
+		versionBeacons := storageMock.NewVersionBeacons(s.T())
+		// Mock the Highest method to return a version beacon with a specific version.
+		versionBeacons.
+			On("Highest", testifyMock.AnythingOfType("uint64")).
+			Return(func(height uint64) (*flow.SealedVersionBeacon, error) {
+				if height == s.height {
+					return &flow.SealedVersionBeacon{
+						VersionBeacon: unittest.VersionBeaconFixture(
+							unittest.WithBoundaries(
+								flow.VersionBoundary{
+									BlockHeight: s.height,
+									Version:     "0.0.1",
+								}),
+						),
+						SealHeight: s.height,
+					}, nil
+				} else {
+					return nil, nil
+				}
+			}, nil)
+
+		var err error
+		s.versionControl, err = version.NewVersionControl(
+			s.log,
+			versionBeacons,
+			semver.New("0.0.1"),
+			s.height-1,
+			s.height,
+		)
+		require.NoError(s.T(), err)
+
+		// Create a mock signaler context for testing.
+		ictx := irrecoverable.NewMockSignalerContext(s.T(), ctx)
+
+		// Start the VersionControl component.
+		s.versionControl.Start(ictx)
+
+		// Ensure the component is ready before proceeding.
+		unittest.RequireComponentsReadyBefore(s.T(), 2*time.Second, s.versionControl)
+
+		// Wait for the asynchronous version controller initialization process.
+		time.Sleep(time.Second)
```

Comment:
is this needed given the previous step waited for the version control system to be ready?